tools/castle-editor/code/framedesign.pas

```diff
@@ -2647,7 +2708,38 @@ procedure TDesignFrame.UpdateSelectedControl;
       for I := 0 to SelectedCount - 1 do
         SelectionForOI.Add(Selected[I]);
       for InspectorType in TInspectorType do
+      begin
         Inspector[InspectorType].Selection := SelectionForOI;
+        if not IsCollectionItemEditFormInitialized then
+        begin
+          for J := 0 to Inspector[InspectorType].RowCount - 1 do
```

Comment:
Looking at the usage of `Continue` and `Break` in this loop, if effectively searches for the 1st row that satisfies `Row.Editor is TCollectionPropertyEditor`, and handles only it. So I would write it more explicitly, as 2 steps,

- find such row
- act on this row

like 

```pascal
Ed := nil;

// search for row with TCollectionPropertyEditor
for J := 0 to Inspector[InspectorType].RowCount - 1 do
begin
  Row := Inspector[InspectorType].Rows[J];
  if Row.Editor is TCollectionPropertyEditor then
  begin
    Ed := TCollectionPropertyEditor(Row.Editor);
    Break;
  end;
end;

// if row with TCollectionPropertyEditor found, act on it
if Ed <> nil then
begin
  Fm := TCollectionPropertyEditorForm(Ed.ShowCollectionEditor(nil, nil, ''));
  ..
end;
```

Reply:
Resolved in 0362e42